File: client-android/lib/src/main/kotlin/stasis/client_android/lib/utils/StringPaths.kt
package stasis.client_android.lib.utils

import java.nio.file.FileSystem

object StringPaths {
    fun String.splitParentAndName(fs: FileSystem): Pair<String, String> {
        val separator = fs.separator
        return when (val i = this.lastIndexOf(separator)) {
            -1 -> Pair("", this)
            0 -> Pair(separator, this.substring(separator.length, this.length))
            else -> Pair(this.substring(0, i), this.substring(i + separator.length, this.length))
        }
    }

    fun String.extractName(fs: FileSystem): String =
        splitParentAndName(fs).second
}

File: client/src/main/scala/stasis/client/utils/StringPaths.scala
package stasis.client.utils

import java.nio.file.FileSystem

object StringPaths {
  implicit class StringAsPath(string: String) {
    def splitParentAndName(fs: FileSystem): (String, String) = {
      val separator = fs.getSeparator

      string.lastIndexOf(separator) match {
        case -1 => ("", string)
        case 0  => (separator, string.substring(separator.length, string.length))
        case i  => (string.substring(0, i), string.substring(i + separator.length, string.length))
      }
    }

    def extractName(fs: FileSystem): String =
      splitParentAndName(fs)._2
  }
}
